### PR TITLE
Fix SDK interactive question showing no question error incorrectly

### LINKED ISSUE #42352: Drill down to see records results in uncaught runtime errors
### Describe the bug

When using drill down "see these <records>", first a "Question not found is shown", then the question actually loads, but a runtime error is triggered.

Loom: https://www.loom.com/share/7530d33a65f145efb0e2547c2c190a1b

image

### To Reproduce

1. Setup Metabase Dockerhub metabase/metabase-dev:embedding-sdk-0.1.0
2. Code is here: https://github.com/albertoperdomo/metabase-sdk-testing-app/commit/f87a757b92b4221c5ccf1b179f807b73b0bda5eb

### Expected behavior

- No runtime errors

### Logs

_No response_

### Information about your Metabase installation

```JSON
- @metabase/embedding-sdk-react v0.1.1
- Metabase Dockerhub metabase/metabase-dev:embedding-sdk-0.1.0
```

### Severity

P1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes https://github.com/metabase/metabase/issues/42352

### Description

The problem happened because we only relied on `loading` which indicates whether the initial question has been loaded or not, after that there could be another loading state when users drills down.

### How to verify

1. Using Customer Zero app.
1. Go to Products > click on a product > Drills on the first bar chart. And there should be no question error state anymore.
1. Click other page on the nav bar e.g. kitchen sink.
1. Click Products again > click on a probuct and there should be no question error state.

#### Before

https://github.com/metabase/metabase/assets/1937582/79cd60ef-60d2-471d-aeac-f0ede2e3299d

#### After
https://www.loom.com/share/65437958d8ff461796e73de630a763b6

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const _InteractiveQuestion = ({
   const hasQuestionChanges =
     card && (!card.id || card.id !== card.original_card_id);
 
-  const [loading, setLoading] = useState(true);
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx:68]: Renaming to make is obvious this means it's loading a question *not* running a question query

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx [hunk 2]
```diff
@@ -99,11 +99,11 @@ export const _InteractiveQuestion = ({
 
   useEffect(() => {
     if (queryResults) {
-      setLoading(false);
+      setIsQuestionLoading(false);
     }
   }, [queryResults]);
 
-  if (loading) {
+  if (isQuestionLoading || isRunning) {
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx:106]: here's the key fix. We should display loading sprinner when either
1. the question is loading
2. the query is running

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx [hunk 3]
```diff
@@ -67,13 +73,16 @@ const setup = ({
...
   setupCardQueryEndpoints(TEST_CARD, TEST_DATASET);
 
-  renderWithProviders(<InteractiveQuestion questionId={TEST_CARD.id} />, {
-    mode: "sdk",
-    sdkConfig: createMockConfig({
-      jwtProviderUri: "http://TEST_URI/sso/metabase",
-    }),
-    storeInitialState: state,
-  });
+  return renderWithProviders(
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx:76]: return the result from `renderWithProvider` so I could access `store` and use it to dispatch actions.

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx [hunk 4]
```diff
@@ -98,6 +107,37 @@ describe("InteractiveQuestion", () => {
...
+      within(screen.getByRole("gridcell")).getByText("Test Row"),
+    ).toBeInTheDocument();
+
+    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+    // Mimicking drilling down by rerunning the query again
+    const storeDispatch = store.dispatch as unknown as ThunkDispatch<
+      State,
+      void,
+      AnyAction
+    >;
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx:130]: I didn't manage to find a way to use this without casting, let me know if there's any better way.

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx [hunk 5]
```diff
@@ -98,6 +107,37 @@ describe("InteractiveQuestion", () => {
...
+    storeDispatch(clearQueryResult());
+    storeDispatch(runQuestionQuery());
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx:132]: These 2 actions do:
1. clear the results so it's as if we're about to query a new question when we're drilling
1. query again

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx [hunk 6]
```diff
@@ -98,6 +107,37 @@ describe("InteractiveQuestion", () => {
     ).toBeInTheDocument();
   });
 
+  it("should render loading state when drilling down", async () => {
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx:110]: This test is tested with the previous version and it failed as expected.

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx [hunk 7]
```diff
@@ -65,7 +65,7 @@ export const _InteractiveQuestion = ({
   const hasQuestionChanges =
     card && (!card.id || card.id !== card.original_card_id);
 
-  const [loading, setLoading] = useState(true);
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
 
   const { isRunning } = uiControls;
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx:70]: I wonder if we should also alias `isRunning` to `isQueryRunning` too, so we can distinguish between question and query.

**Review by heypoom — APPROVED** [review]
The fix works on the demo app, nice! Verified that all three cases (drill-down, reset view, navigate back and forth) does not result in a Question Not Found error.

The renaming is a good idea too - I like that it semantically distinguishes query vs question. 

(Ignore the height of the second container)


https://github.com/metabase/metabase/assets/4714175/a5b6a694-681d-4b41-ab57-31427402ff7a

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx (hunk 7 above)
Comment (WiNloSt) [reply enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx:70]: Good idea.

Comment (WiNloSt) [reply enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx]: Fixed in 4654ee28410d938452c7a5127e7861388f78a6f2

**Pull request merged**